### ISSUE TITLE
Brainstorm new multiplayer friendly story

### ISSUE THREAD

**Issue by Sunrunner37** [SubnauticaNitrox/Nitrox#138]

**Comment by IsraBrine** [comment]
I mean, once you get that far, if you make it like a 2 player co-op only, you could probably go with the fact that there were two seats in Life-pod 5, yet there is the problem of figuring out how to knockout both players in the opening sequence. Like is maybe the metal bit hits one then bounces off to the other. Then there's the whole fire bit that occurs as soon as you get out of the seat, maybe one person (probably the host of the game) get out of the first seat while the other person is still stuck in his seat, so once the host finishes putting out the fire, maybe there could be an animation for the host helping the second player out of their seat. I don't know if this is what you meant by the topic there, but this is just my input.

**Comment by Dehydrated-Water** [comment]
Maybe one player could be hit with the fire extinguisher, or already be unconscious before boarding, and player one dragged them on? I think the idea of only the host putting out the fire then an animation of helping the other player out of his/her seat is good.
I would also be willing (and eager) to help with story changes.
I'm excited for what's to come!
-Cheers, Dehydrated-Water

**Comment by Dehydrated-Water** [comment]
The Neptune Rocket only carries one passenger, so, you would need a different rocket, or a modification of the Neptune.
In either case, it must be considered where the two seats will be placed, they could be;
Next to eachother - This offers a sense of the two players being a team, as you would be able to see the 
                                 other player's shoulder at the side of your camera when you turned to the 
                                 left/right, while not compromising the view of the planet as you take off and move 
                                 away from it. [This placement gives both a sense of comradery and the view.]                                 
Facing eachother - This completely compromises the view of the plant for one player, as they would be 
                               facing away from the planet, while the other player sees the planet go by, with the 
                               other player inbetween them and that view. [This seems to offer the highest sense of 
                               togetherness, (and also possible nostalgia, since, in the life pod, both players are 
                               facing eachother.) But lacks on the view aspect. However, if the players were 
                               positioned so that neither one faced the planet, (they both faced eachother and were 
                               side-on to the planet, it could be possible to have the ability to look left, and see the 
                               planet whizzing past, and look right, and see the stars. This view has the most 
                               potential.)]
or
Back to back - This is probably the worst as it offers no sense of teamwork. This could, however, be 
                        employed if things run sour between the two survivors, or if a together-like ending was not desired.

**Comment by thedestroyer81624** [comment]
I think the intro isn’t needed for nitrox. True, rocket and disease scenes should be included, but others could be left out.

**Comment by V1r1d1s** [comment]
Maybe for the start it would be easier if you would do it like the forest where the story and the sequences are still made for one player, but you can do everything with multiples...
I personaly don't like the idea of making a whole new story just because "it doesn't fit anymore". But afterall thats just a personal opinion and you guys are the beatifull people who make this mod... ;-)

**Comment by Dehydrated-Water** [comment]
Yeah it would certainly be easier, and more authentic. Either way I think a couple cut-scenes with both characters would still help to set the tone and mood, which are two of the things Subnautica does so well. To sum up, even if we don't end up coming up with a new story, I think a couple scenes with both players could be really beneficial to atmosphere.

**Comment by V1r1d1s** [comment]
But what if you play with more than two players? doing something scalable would be very hard I think...
Another idea would be to make the cutscenes from a perspective where you can't realy see how many players there are. (Like an outside view from the Droppod coming down from the aurora and then just the player coming out of that pod.)

**Comment by CalzLight** [comment]
you could make changes to the sea emperors speech to fit two people better, as well as when the PDA talks it could maybe suggest helping a friend if they pass out due to oxygen loss in a kind of revival type method.

**Comment by Dehydrated-Water** [comment]
Very good ideas. I'll take a look at the Sea Emperor's speech and see what can be changed without impacting the storyline.

**Comment by Dehydrated-Water** [comment]
@V1r1d1s You're definitely correct there. The scenes should be engineered to include minimal players

**Comment by V1r1d1s** [comment]
I think also from a gameplay perspective it would make scene to have a new life pod for each new (or each 2 new) players. I mean like MedKit spawn rate and shared storage...

**Comment by NonoTCC** [comment]
I think that there should be two separate story's, one for 2 players, and one for if you want to have 4 or 3 players play together. The 2 player story is that you have both players in one lifepod, host is hit by a fire extinguisher, the other person (in the seat that isn't taken in single-player) manages to not get hit by the extinguisher, and it knocks the metal panel off the wall. player 2 rescues player 1, and story continues as normal. (Except for using Dehydrated Water's Neptune Rocket seating and Sea Emperor line changes.) 4 Player is that the 4 are friends (Or not) and that they take different lifepods, both crash in Safe Shallows (or not, again), and need to survive. 3 player could be that in the other lifepod, the 4th person gets killed by the metal plate. There could be more complexity in there, but those are my ideas.

**Comment by disseminate** [comment]
> I had originally imagined that additional players would spawn in their own life pods, kind of like how pod 17 and pod 8(?) Had made it off the ship and landed on the planet.
> 
> I had just imagined that each player had also landed in their own life pods also.
> 
> How hard would it be to have a new life pod spawn within a certain range of life pod 5 for each new player (or after each 2 new players) that joins the server after 2?
> 
> It would make the player spawning scalable, and it could be interesting to start in new ( although relatively close) locations.
> 
> If life pod 5's asset is used for each new life pod, it should just be a matter of writing the code to spawn in another life pod 5 in a different location, right?
> 
> That's all I have for now. I am a writer in my spare time though, so once I have finished the story line for Subnautica I will likely have more to contribute.

Right now there's some boilerplate code to spawn in one pod per 50 players. They're spawned in in a line in one direction, very close to each other. That being said it should be possible to make it seem more organic in positioning and quantity - my concern is that there could be some misc. game rules hardcoded into there just being one pod.

**Comment by Dehydrated-Water** [comment]
> I think also from a gameplay perspective it would make scene to have a new life pod for each new (or each 2 new) players. I mean like MedKit spawn rate and shared storage...

I agree, you would need to have a reasonable amount of Medkits

**Comment by Jannify** [comment]
I have read all the comments and want to update a little on our progress and plans:

Very soon we will have a lifepod cinematic with two players, both getting knocked out in the progress. The animation is more or less mirrored, so they wake up simultaneously and can both put out the fires. The idea with one player doing the fire stuff and rescuing the other player after that is a neat idea but needs more thought as we are more or less binding and blocking one player to the actions of another.

Multiple life pods are currently not implemented but they will come. We plan to use one pod per 1-2 simultaneously joining players (and playing one of the two cinematics). But this needs to be optional to make big servers feasible.

We also had the idea to let the pod of joining player fall out of the sky, so players already connected can see them. But that is something for the future I believe.

The idea with the subtle change of the emperor monologue is perfect. I really want to implement that as I think that changes like this really add to the immersion.
